### DISCUSSION TITLE
不同的操作系统，build的文件hash不一致，已确认node版本一致

### DISCUSSION THREAD

**Discussion by chenyu1990** [umijs/umi#10357] [Q&A]
我需要上传到oss。在不同操作系统上编译上传，因为build的文件hash不一致，导致冗余了很多文件。

**Comment by fz6m** [comment]
不同操作系统安装的依赖是不一样的，很多 npm 包会根据当前平台下载对应的平台包，同时压缩时随机生成的字符不一定每次都是一样的，另外不同系统的文件结尾和换行也有区别，所以产物应该仅在某个系统编译后上传一份。

**Comment by xiaohuoni** [comment]
不清空旧的文件夹吗？

**Reply by chenyu1990** [reply]
> 不清空旧的文件夹吗？

对比上传可以省流
